packages/react/src/components/hooks/useWebRTC.ts

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { WebRTC } from "@signalwire/js";
+
+export default function useWebRTC(deviceEnabled: {
+  cameras: boolean;
+  microphones: boolean;
+  speakers: boolean;
+}) {
+  const [cameras, setCameras] = useState<MediaDeviceInfo[]>([]);
+  const [microphones, setMicrophones] = useState<MediaDeviceInfo[]>([]);
+  const [speakers, setSpeakers] = useState<MediaDeviceInfo[]>([]);
+
+  useEffect(() => {
+    async function setupList() {
+      if (deviceEnabled.cameras !== false) {
+        const cameras = await WebRTC.getCameraDevicesWithPermissions();
+        setCameras(Array.from(cameras));
+      }
+      if (deviceEnabled.microphones !== false) {
+        const microphones = await WebRTC.getMicrophoneDevicesWithPermissions();
+        setMicrophones(microphones);
+      }
+      if (deviceEnabled.speakers !== false) {
+        const speakers = await WebRTC.getSpeakerDevicesWithPermissions();
+        setSpeakers(speakers);
+      }
```

Comment:
we are awaiting for cameras, mics, and speakers in sequence, but these could actually be done in parallel

Reply:
Hm, makes sense. Changing ...

Reply:
also, the `*withPermissions` methods are deprecated. Prefer using the variants without `withPermissions`